cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +741,98 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath).replace("\r", ""))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
+
+    if userPath:
+        ud = util.load_file(userPath).replace("\r", "")
+    return md, ud, cfg, network
+
+
+class LoadError(Exception):
```

Comment:
If you change the loader to just load with yaml or json, then you can drop this.  It doesn't actually provide us with any additional information , in fact your use of it *loses* information as the original error had info on what was invalid (json parse or yaml parse).

Reply:
Fixed